lib/admin/WP_Auth0_Admin_Advanced.php

```diff
@@ -134,294 +134,233 @@ public function render_passwordless_method() {
+    );
   }
 
-  public function render_auto_login() {
-    $v = absint( $this->options->get( 'auto_login' ) );
-
-    echo $this->render_a0_switch( "wpa0_auto_login", "auto_login", 1, 1 == $v );
-?>
-
-    <div class="subelement">
-      <span class="description"><?php echo __( 'Mark this to avoid the login page (you will have to select a single login provider)', 'wp-auth0' ); ?></span>
-    </div>
-    <?php
+  public function render_auto_login( $args ) {
+    $this->render_switch( $args[ 'label_for' ], $args[ 'opt_name' ] );
+    $this->render_field_description(
+      __( 'Send logins directly to a specific Connection, skipping the login page', 'wp-auth0' )
+    );
   }
 
-  public function render_auto_login_method() {
-    $v = $this->options->get( 'auto_login_method' );
-?>
-    <input type="text" name="<?php echo $this->options->get_options_name(); ?>[auto_login_method]" id="wpa0_auto_login_method" value="<?php echo esc_attr( $v ); ?>"/>
-    <div class="subelement">
-      <span class="description"><?php echo __( 'To find the method name, log into Auth0 Dashboard, and navigate to: Connection -> [Connection Type] (eg. Social or Enterprise). Click the "down arrow" to expand the wanted method, and use the value in the "Name"-field. Example: google-oauth2', 'wp-auth0' ); ?></span>
-    </div>
-    <?php
+  public function render_auto_login_method( $args ) {
+    $this->render_text_field( $args[ 'label_for' ], $args[ 'opt_name' ] );
+    $this->render_field_description(
+      __( 'Find the method name to use under Connections > [Connection Type] in your ', 'wp-auth0' ) .
+      $this->get_dashboard_link( 'connections', TRUE ) .
+      __( 'Click the expand icon and use the value in the "Name" field (like "google-oauth2")', 'wp-auth0' )
+    );
   }
 
-  public function render_ip_range_check() {
-    $v = absint( $this->options->get( 'ip_range_check' ) );
-
-    echo $this->render_a0_switch( "wpa0_ip_range_check", "ip_range_check", 1, 1 == $v );
+  public function render_ip_range_check( $args ) {
+    $this->render_switch( $args[ 'label_for' ], $args[ 'opt_name' ] );
   }
 
-  public function render_ip_ranges() {
-    $v = $this->options->get( 'ip_ranges' );
-?>
-    <textarea cols="25" name="<?php echo $this->options->get_options_name(); ?>[ip_ranges]" id="wpa0_ip_ranges"><?php echo esc_textarea( $v ); ?></textarea>
-    <div class="subelement">
-      <span class="description"><?php echo __( 'Only one range per line! Range format should be as follows (spaces will be trimmed):', 'wp-auth0' ); ?></span>
-      <code>xx.xx.xx.xx - yy.yy.yy.yy</code>
-    </div>
-    <?php
+  public function render_ip_ranges( $args ) {
+    $this->render_textarea_field( $args[ 'label_for' ], $args[ 'opt_name' ] );
+    $this->render_field_description(
+      __( 'Only one range per line! Range format should be as follows (spaces ignored): ', 'wp-auth0' ) .
+      __( '<br><code>xx.xx.xx.xx - yy.yy.yy.yy</code>', 'wp-auth0' )
```

Comment:
`<br>` doesn't require a closure tag?

Reply:
[Not since HTML5](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/br)